addOns/sequence/src/main/java/org/zaproxy/zap/extension/sequence/StdActiveScanRunner.java

```diff
@@ -0,0 +1,165 @@
+                Control.getSingleton().getExtensionLoader().getExtension(ExtensionSequence.class);
+
+        this.name = wrapper.getName();
+        fakeRoot = new SiteNode(null, HistoryReference.TYPE_SEQUENCE_TEMPORARY, name);
+        fakeDirectory = new SiteNode(null, HistoryReference.TYPE_SEQUENCE_TEMPORARY, name);
+        fakeRoot.add(fakeDirectory);
+        step = 0;
+    }
+
+    @Override
+    public String run(ZestScript script, Map<String, String> params)
+            throws ZestAssertFailException,
+                    ZestActionFailException,
+                    IOException,
+                    ZestInvalidCommonTestException,
+                    ZestAssignFailException,
+                    ZestClientFailException {
+        return super.run(this.wrapper.getZestScript(), params);
+    }
+
+    @Override
+    public ZestResponse runStatement(
+            ZestScript script, ZestStatement stmt, ZestResponse lastResponse)
+            throws ZestAssertFailException,
+                    ZestActionFailException,
+                    ZestInvalidCommonTestException,
+                    IOException,
+                    ZestAssignFailException,
+                    ZestClientFailException {
+        ZestResponse resp = super.runStatement(script, stmt, lastResponse);
+
+        if (stmt instanceof ZestRequest) {
+            step += 1;
+            ZestRequest req = (ZestRequest) stmt;
+            HttpMessage msg = ZestZapUtils.toHttpMessage(req, req.getResponse());
+            SiteNode node = messageToSiteNode(msg, step);
+            if (node != null) {
+                fakeDirectory.add(node);
+
+                int scanId =
+                        extSeq.getExtActiveScan()
+                                .startScan(new Target(node), user, new Object[] {scanPolicy});
+
+                ActiveScan ascan = extSeq.getExtActiveScan().getScan(scanId);
+                while (ascan.isRunning()) {
+                    try {
+                        Thread.sleep(1000);
+                    } catch (InterruptedException e) {
+                        // Ignore
+                    }
+                }
+            }
+        }
+
+        return resp;
+    }
+
+    private SiteNode messageToSiteNode(HttpMessage msg, int step) {
+        SiteNode temp = null;
+        try {
+            temp =
+                    new SiteNode(
+                            null,
+                            SEQUENCE_HISTORY_TYPE,
+                            Constant.messages.getString("sequence.automation.step", step));
+            HistoryReference ref =
+                    new HistoryReference(
+                            extHistory.getModel().getSession(), SEQUENCE_HISTORY_TYPE, msg);
+
+            extHistory.addHistory(ref);
+            // These tags will propagate to any alerts raised against them
+            ref.addTag("ALERT-TAG:zap-seq-name=" + wrapper.getName());
+            ref.addTag("ALERT-TAG:zap-seq-index=" + step);
```

Comment:
The ALERT-TAG part gets dropped right? Should the comment call that out?

Reply:
can do..